packages/postgresql-node-connector/src/index.ts

```diff
@@ -36,7 +36,7 @@ export class PostgresServiceFactory implements Node.ServiceFactory {
   private connection: Connection;
 
   constructor(
-    configuration: ConnectionOptions,
+    readonly configuration: ConnectionOptions | PostgresServiceFactory,
```

Comment:
Is this the kind of union you were talking about @cf19drofxots ?

Reply:
Oh wow, autocomplete got me. Let me fix this lol

Reply:
Ah not exactly. More like, for the actual `PostgresConnectionOptions`, converting the interface to a type, like:
```
export type PostgresConnectionOptions = ConnectionOptions | { <the-current-properties> };
```

Reply:
And then passing this updated `PostgresConnectionOptions` to this constructor.